fe/fe-core/src/main/java/com/starrocks/scheduler/PartitionBasedMaterializedViewRefreshProcessor.java

```diff
@@ -518,11 +522,18 @@ public Set<String> getPartitionsToRefreshForMaterializedView(Map<String, String>
         String start = properties.get(TaskRun.PARTITION_START);
         String end = properties.get(TaskRun.PARTITION_END);
         boolean force = Boolean.parseBoolean(properties.get(TaskRun.FORCE));
+        Set<String> needRefreshMvPartitionNames = Sets.newHashSet();
+        PartitionInfo partitionInfo = materializedView.getPartitionInfo();
+        if (this.mvContext.status != null) {
```

Comment:
I think you can get partition star and partition end at the end of this function, getPartitionNamesByRangeWithPartitionLimit may change the partition star and partition end

Reply:
fix it.